android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java

```diff
@@ -1095,16 +1112,36 @@ public void run() {
             }
             WritableMap data = mRNCWebViewClient.createWebViewEvent(webView, webView.getUrl());
             data.putString("data", message);
-            dispatchEvent(webView, new TopMessageEvent(webView.getId(), data));
+
+            if (mCatalystInstance != null) {
```

Comment:
Could you create another method to remove the duplicated code?

Reply:
I wouldn't worry about the duplicated code in this instance. It's a pretty minor bit of duplication, close together.

Reply:
I tried to keep the additions close to existing code there.